include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h

```diff
@@ -44,6 +54,97 @@ class __radix_sort_reorder_peer_kernel;
 template <::std::uint32_t, bool, bool, typename... _Name>
 class __radix_sort_reorder_kernel;
 
+//------------------------------------------------------------------------
+// radix sort: bitwise order-preserving conversions to unsigned integrals
+//------------------------------------------------------------------------
+
+#if __cpluslplus >= 202002L && __has_include(<bit>)
+template <typename _Dst, typename _Src>
+using __dpl_bit_cast = std::bit_cast<_Dst, _Src>;
+
+#elif __has_builtin(__builtin_bit_cast)
+template <typename _Dst, typename _Src>
+_Dst
+__dpl_bit_cast(const _Src& __src)
+{
+    static_assert(sizeof(_Dst) == sizeof(_Src), "Bit conversion for types of different size");
+    return __builtin_bit_cast(_Dst, __src);
+}
+
+#else
+template <typename _Dst, typename _Src>
+_Dst
+__dpl_bit_cast(const _Src& __src)
+{
+    static_assert(sizeof(_Dst) == sizeof(_Src), "Bit conversion for types of different size");
```

Comment:
Maybe also add
```c++
static_assert(std::is_trivially_copyable_v<_Dst> && std::is_trivially_copyable_v<_Src>);
```

Reply:
I would suggest definitely add this but the question is where. I fill like it should be constraint-like (enable_if basically) rather than `static_assert`. Probably even for `sizeof` case. By the way, after expressing my filling I've checked STD requirements and the are doing like "Overload should not participate in overload resolution..." basically meaning `enable_if`